File: 251004/Yelkin/4-5/CommentService/src/main/java/by/yelkin/CommentService/controller/CommentController.java
package by.yelkin.CommentService.controller;

import by.yelkin.CommentService.dto.CommentRequestTo;
import by.yelkin.CommentService.dto.CommentResponseTo;
import by.yelkin.CommentService.service.CommentService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;


@RestController
@RequestMapping("/api/v1.0/comments")
public class CommentController {

    private final CommentService commentService;

    @Autowired
    public CommentController(CommentService commentService) {
        this.commentService = commentService;
    }


    @GetMapping
    public ResponseEntity<List<CommentResponseTo>> getAllComments() {
        List<CommentResponseTo> posts = commentService.findAll();
        return ResponseEntity.ok(posts);
    }

    @GetMapping("/{id}")
    public ResponseEntity<CommentResponseTo> getCommentsById(@PathVariable Long id) {

        CommentResponseTo responseTo = commentService.findById(id);
        return ResponseEntity.ok(responseTo);
    }

    @PutMapping("/{id}")
    public ResponseEntity<CommentResponseTo> update(@PathVariable(required = false) Long id, @RequestBody CommentRequestTo commentRequestTo) {
        if (id != null) {
            commentRequestTo.setId(id);
        }
        return ResponseEntity.ok(commentService.update(commentRequestTo));
    }

}

File: 251003/Kukhatskavolets/microblog/discussion/src/main/kotlin/by/kukhatskavolets/discussion/controllers/CommentController.kt
package by.kukhatskavolets.discussion.controllers

import by.kukhatskavolets.discussion.dto.requests.CommentRequestTo
import by.kukhatskavolets.discussion.dto.responses.CommentResponseTo
import by.kukhatskavolets.discussion.services.CommentService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1.0/comments")
class CommentController(
    private val commentService: CommentService
) {
    @PostMapping
    fun createComment(
        @RequestBody @Valid commentRequestTo: CommentRequestTo,
        @RequestHeader("X-Country", defaultValue = "Belarus") country: String
    ): ResponseEntity<CommentResponseTo> {
        val createdComment = commentService.createComment(commentRequestTo, country)
        println(createdComment)
        return ResponseEntity(createdComment, HttpStatus.CREATED)
    }

    @GetMapping
    fun getAllComments(): ResponseEntity<List<CommentResponseTo>> {
        val comments = commentService.getAllComments()
        return ResponseEntity.ok(comments)
    }

    @GetMapping("/{id}")
    fun getCommentById(
        @PathVariable id: Long,
        @RequestHeader("X-Country", defaultValue = "Belarus") country: String
    ): ResponseEntity<CommentResponseTo> {
        val comment = commentService.getCommentById(country, id)
        return ResponseEntity.ok(comment)
    }

    @DeleteMapping("/{id}")
    fun deleteComment(
        @PathVariable id: Long,
        @RequestHeader("X-Country", defaultValue = "Belarus") country: String
    ): ResponseEntity<Void> {
        commentService.deleteComment(country, id)
        return ResponseEntity.noContent().build()
    }

    @PutMapping()
    fun updateComment(
        @RequestBody @Valid commentRequestTo: CommentRequestTo,
        @RequestHeader("X-Country", defaultValue = "Belarus") country: String
    ): ResponseEntity<CommentResponseTo> {
        val updatedComment = commentService.updateComment(country, commentRequestTo)
        return ResponseEntity.ok(updatedComment)
    }
}